### PR TITLE
Erlang/OTP 21 logger integration

### PR DESCRIPTION
**Pull request by josevalim**

### CODE REVIEW

**Comment by michalmuskala** [comment]
Could we get the optimisations to `Enum` and `Keyword` in separate commit/PR? I think this will help keep the discussion focused.

**Comment by josevalim** [comment]
This is ready for review. I just need to write tests for the new handler and it is ready to go.

File: lib/logger/lib/logger/erlang_handler.ex [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+      end
+    end)
+  end
+
+  defp erlang_level_to_elixir_level(:emergency), do: :error
+  defp erlang_level_to_elixir_level(:alert), do: :error
+  defp erlang_level_to_elixir_level(:critical), do: :error
+  defp erlang_level_to_elixir_level(:error), do: :error
+  defp erlang_level_to_elixir_level(:warning), do: :warn
+  defp erlang_level_to_elixir_level(:info), do: :info
```
Comment (sirihansen) [inline lib/logger/lib/logger/erlang_handler.ex]: Missing `notice`.

Comment (josevalim) [reply lib/logger/lib/logger/erlang_handler.ex]: @sirihansen Thanks! I am glad you *noticed* that.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Comment by josevalim** [comment]
I have merged this because it causes some intermittent failures in our suite, making it really hard to work on other Erlang/OTP 21 features. Please drop any other feedback in this thread and I will get to it. :)

**Comment by sirihansen** [comment]
Just one minor doc-thing - logger.ex, line 574 - should probably remove "both":

  ```
* `:flush` - when `true`, guarantees all messages currently sent
  +      to both `Logger` are processed before the backend is removed
```

**Comment by josevalim** [comment]
@sirihansen thanks!

**Pull request merged**